localstack/aws/protocol/service_router.py

```diff
@@ -374,29 +375,39 @@ def determine_aws_service_name(request: Request, services: ServiceCatalog = None
         return None
 
     # 5. check the query / form-data
-    values = request.values
-    if "Action" in values:
-        # query / ec2 protocol requests always have an action and a version (the action is more significant)
-        query_candidates = [
-            service
-            for service in services.by_operation(values["Action"])
-            if services.get(service).protocol in ("ec2", "query")
-        ]
-
-        if len(query_candidates) == 1:
-            return query_candidates[0]
-
-        if "Version" in values:
-            for service in list(query_candidates):
-                service_model = services.get(service)
-                if values["Version"] != service_model.api_version:
-                    # the combination of Version and Action is not unique, add matches to the candidates
-                    query_candidates.remove(service)
-
-        if len(query_candidates) == 1:
-            return query_candidates[0]
-
-        candidates.update(query_candidates)
+    try:
+        values = request.values
+        if "Action" in values:
+            # query / ec2 protocol requests always have an action and a version (the action is more significant)
+            query_candidates = [
+                service
+                for service in services.by_operation(values["Action"])
+                if services.get(service).protocol in ("ec2", "query")
+            ]
+
+            if len(query_candidates) == 1:
+                return query_candidates[0]
+
+            if "Version" in values:
+                for service in list(query_candidates):
+                    service_model = services.get(service)
+                    if values["Version"] != service_model.api_version:
+                        # the combination of Version and Action is not unique, add matches to the candidates
+                        query_candidates.remove(service)
+
+            if len(query_candidates) == 1:
+                return query_candidates[0]
+
+            candidates.update(query_candidates)
+
+    except RequestEntityTooLarge:
+        # Some requests can be form-urlencoded but also contain binary data, which will fail the form parsing (S3 can
+        # do this). In that case, skip this step and continue to try to determine the service name.
+        LOG.debug(
+            "Failed to parse the form while determining AWS service name",
+            exc_info=LOG.isEnabledFor(logging.DEBUG),
+        )
+        pass
```

Comment:
```suggestion
        LOG.debug(
            "Failed to determine AWS service from request body because payload was too large",
            exc_info=LOG.isEnabledFor(logging.DEBUG),
        )
```

Reply:
Hmm, I'm not too sure about this because the error was not that the payload was too large, but that it contained binary data. Werkzeug will however raise a 413 exception which indicate the payload was too large, but it was a red herring. Which is why I wanted to log a form parsing failure instead of the size error. Does it make sense?

Reply:
@thrau could `Failed to determine AWS service from request body because the form could not be parsed` be a middle ground?